CommonMark/Parser/TableMethods.cs

```diff
@@ -0,0 +1,365 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using CommonMark.Syntax;
+
+namespace CommonMark.Parser
+{
+	internal static class TableMethods
+	{
+		static List<string> ParseTableLine(StringPart part, StringBuilder sb)
+		{
+			if (part.Source == null)
+				return null;
+
+			string line = part.Source.Substring(part.StartIndex, part.Length);
+			line = line.TrimEnd('\n');
+
+			var ret = new List<string>();
+
+			var i = 0;
+
+			if (i < line.Length && line[i] == '|') i++;
+
+			while (i < line.Length && char.IsWhiteSpace(line[i])) i++;
+
+			for (; i < line.Length; i++)
+			{
+				var c = line[i];
```

Comment:
This should be rewritten using line.IndexOfAny(new [] { '\\', '|' } ), however for now that could be added as a `//TODO: ` comment here and the code stays as is to get it merged sooner.